src/vision/status_quo/barbara_simulates_hydrodynamics.md

```diff
@@ -0,0 +1,73 @@
+```rust
+Ok::<_, HydroError>( ( p.to_shared(), s.to_shared() ) )  
+```
+And she could not figure out why she had to add the `::<_, HydroError>` to some of the `Result` values.
+
+Once her team began using the new `async` design for their simulations, they noticed an important issue that impacted productivity: compilation time had now increased to between 30 and 60 seconds. The nature of their work requires frequent changes to code and recompilation and 30-60 seconds is long enough to have a noticeable impact on their quality of life.  What her and her team want is for compilation to be 2 to 3 seconds. Barbara believes that the use of `async` is a major contributor to the long compilation times.
+
+This new solution works, but Barbara is not satisfied with how complex her code wound up becoming with the move to `async` and the fact that compilation time is now 30-60 seconds.  The state sharing adding a large amount of cruft with `Arc` and `async` is not well suited for using a dependency graph to schedule tasks so implementing this solution created a key component of her program that was difficult to understand and pervasive. Ultimately, her conclusion was that `async` is not appropriate for parallelizing computational tasks. She will be trying a new design based upon Rayon in the next version of her application.
+
+## 🤔 Frequently Asked Questions
+
+### **What are the morals of the story?**
+- `async` looks to be the wrong choice for parallelizing compute bound/computational work
+- There is a lack of guidance to help people solving such problems get started on the right foot
+- Quality of Life issues (compilation time, type inference on `Result`) can create a drag on users ability to focus on their domain problem
+
+### **What are the sources for this story?**
+This story is based on the experience of building the [kilonova](https://github.com/clemson-cal/app-kilonova) hydrodynamics simulation solver.
+
+### **Why did you choose *NAME* to tell this story?**
+I chose Barbara as the primary character in this story because this work was driven by someone with experience in Rust specifically but does not have much systems level experience. Grace was chosen as a supporting character because of that persons experience with C/C++ programming and to avoid repeating characters.
+
+### **How would this story have played out differently for the other characters?**
+For Alan, there's a good chance he would have already had experience working with either async workflows in another language or doing parallelization of compute bound tasks; and so would already know from experience that `async` was not the right place to start.  Grace, likewise, might already have experience with problems like this and would know what to look for when searching for tools. For Niklaus, the experience would probably be the same, as it's very easy to assume that `tokio` is the starting place for concurrency in Rust.
```

Comment:
I think using a bullet list for the different characters is better readable and lines up with the current style

Reply:
Fixed.